### PR TITLE
Create github actions for Continuous Integration!

### PR DESCRIPTION
**Pull request by abhijitvalluri**
While Travis is good, we can also leverage Github actions for built in CI within the Github environment and UI. You can see the status of these checks directly on Github through the "Checks" tab on the PR. Also, we can configure multiple checks to run separately. Hence, due to this parallelism, the checks should finish quicker.

Currently, I enabled the following checks:
* build - builds the forkRelease version of the app
* tests - runs the Android tests
* detekt - runs the detekt checks for kotlin
* ktlint - runs kotlin lint checks
* lint - creates a lint result 

These checks run on each PR and should run on the latest commit on the fork (default) branch as well.
I have also configured these checks to run daily at 12 PM UTC time.

Also, I have made a lot of changes to the code to fix existing lint errors, and detekt errors and also suppress errors that cannot be fixed trivially and are not worth the hassle of fixing.

### CODE REVIEW

**Comment by abhijitvalluri** [comment]
@interfect please take a look! :) If we are happy with Github actions, we can get rid of Travis CI. OR we can keep both. But the Github actions duplicate the checks in Travis CI and hence make it redundant.

I feel that having all the various checks running separately via Github actions makes it easier for us to re-run any specific check if it fails erroneously. However, currently in Travis all the checks run in one workflow. So, if any one of those checks fail we will have to redo ALL checks and it can take a long time. 

Also because the checks in Travis (currently running tests and building the app) run one after the other, the overall time to finish them can be quite high. But the Github actions run separately, so it should take lesser time to finish the checks.

**Comment by interfect** [comment]
This does look nicer than Travis. I'm not sure how long Github is going to want to pay to run our stuff, but it also looks like we can use more compute than on Travis.

**Comment by interfect** [comment]
Is there a point to doing daily builds if we're also doing builds on every PR we take? If nothing has changed it might be a waste.

**Comment by abhijitvalluri** [comment]
> Is there a point to doing daily builds if we're also doing builds on every PR we take? If nothing has changed it might be a waste.

We can disable the daily builds, but typically people run tests on a daily basis to make sure that the project is in a good state. As long as Github lets us do it I think it is a nice extra to have. But upto you.

Also, should I disable Travis as the Github actions duplicate Travis tests and of course go further than that?

> This does look nicer than Travis. I'm not sure how long Github is going to want to pay to run our stuff, but it also looks like we can use more compute than on Travis.

Github seems to be happy to offer Github actions for free accounts as well with some limitations on usage. I hope they will continue to offer this forever, but if they don't we could go back to Travis at that time I suppose.

**Review by interfect — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/components/topsheet/TopSheetBehavior.kt [hunk 1]
```diff
@@ -806,5 +821,7 @@ class TopSheetBehavior<V : View?>
         private const val CORNER_ANIMATION_DURATION = 500
         private val DEF_STYLE_RES = R.style.Widget_Design_BottomSheet_Modal
 
+        private const val PEEK_HEIGHT_RATIO = 3 / 4
```
Comment (abhijitvalluri) [inline app/src/main/java/org/mozilla/fenix/components/topsheet/TopSheetBehavior.kt:824]: This has caused a bug! 3/4 becomes 0! This causes the top tabs tray from not appearing when in portrait mode. Fixing in new PR.

**Pull request merged**